pysmt/solvers/optimizer.py

```diff
@@ -0,0 +1,313 @@
+#
+# This file is part of pySMT.
+#
+#   Copyright 2014 Andrea Micheli and Marco Gario
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from pysmt.solvers.solver import Solver
+from pysmt.exceptions import PysmtValueError
+
+class Optimizer(Solver):
+    """
+    Interface for the optimization
+    """
+
+    def optimize(self, cost_function, **kwargs):
```

Comment:
Why isn't this called `minimize`? `optimize` does not make it clear in which direction we are working (min/max).